query/physicalplan/physicalplan.go

```diff
@@ -14,10 +15,12 @@ import (
 	"github.com/polarsignals/frostdb/query/logicalplan"
 )
 
+var concurrency = runtime.NumCPU()
+
 type PhysicalPlan interface {
-	Callback(ctx context.Context, r arrow.Record) error
-	Finish(ctx context.Context) error
+	Callbacks() []func(ctx context.Context, r arrow.Record) error
```

Comment:
Maybe we should consider making callback a type so that we don't have to keep typing it out

Reply:
+1, or an interface